### PR TITLE
[PWGHF] Omegac0/Xi pT cut

### PR DESCRIPTION
**Pull request by FDUEnrich**
Calculating pT of Omegac0/Xi and cut pT< setting value. Fill after pTcut distribution histos and pTcut status.

### CODE REVIEW

File: PWGHF/TableProducer/trackIndexSkimCreator.cxx [hunk 1]
```diff
@@ -3234,7 +3235,11 @@ struct HfTrackIndexSkimCreatorLfCascades {
     // Selection criteria
     // selections have been set to run2 lambda dedicated cuts
     // selections for cascade have been set to the loosest value between xi and omega
-    // a tolerance has been added to be more conservative
+    // a tolerance has been added to be more conservative  ptMinOmegaczeroToOmegaKaLfCasc ptMinXicZeroOmegacZeroToXiPiLfCasc
+    Configurable<float> ptMinOmegacZeroToOmegaPiLfCasc{"ptMinOmegacZeroToOmegaPiLfCasc", 2.0, "min. pT for Omegaczero in Omega + Pi decays"};
+    Configurable<float> ptMinOmegaczeroToOmegaKaLfCasc{"ptMinOmegaczeroToOmegaKaLfCasc", 2.0, "min. pT for Omegaczero in Omega + Ka decays"};
+    Configurable<float> ptMinXicZeroOmegacZeroToXiPiLfCasc{"ptMinXicZeroOmegacZeroToXiPiLfCasc", 2.0, "min. pT for XicZeroOmegacZero in Xi + Pi decays"};
+    Configurable<float> ptMinXicplusLfCasc{"ptMinXicplusLfCasc", 2.0, "min. pT for Xicplus in Xi + Pi + Pi decays"};
```
Comment (fgrosa) [inline PWGHF/TableProducer/trackIndexSkimCreator.cxx:3242]: I would put 0 as default not to change the behaviour of pp wagons
```suggestion
    // a tolerance has been added to be more conservative  ptMinOmegaczeroToOmegaKaLfCasc ptMinXicZeroOmegacZeroToXiPiLfCasc
    Configurable<float> ptMinOmegacZeroToOmegaPiLfCasc{"ptMinOmegacZeroToOmegaPiLfCasc", 0.f, "min. pT for Omegaczero in Omega + Pi decays"};
    Configurable<float> ptMinOmegaczeroToOmegaKaLfCasc{"ptMinOmegaczeroToOmegaKaLfCasc", 0.f, "min. pT for Omegaczero in Omega + Ka decays"};
    Configurable<float> ptMinXicZeroOmegacZeroToXiPiLfCasc{"ptMinXicZeroOmegacZeroToXiPiLfCasc", 0.f, "min. pT for XicZeroOmegacZero in Xi + Pi decays"};
    Configurable<float> ptMinXicplusLfCasc{"ptMinXicplusLfCasc", 0.f, "min. pT for Xicplus in Xi + Pi + Pi decays"};
```

File: PWGHF/TableProducer/trackIndexSkimCreator.cxx [hunk 2]
```diff
@@ -3555,17 +3570,23 @@ struct HfTrackIndexSkimCreatorLfCascades {
               std::array<float, 3> pVecPion1XiHyp = {0.};
               df2.getTrack(0).getPxPyPzGlo(pVecXi);
               df2.getTrack(1).getPxPyPzGlo(pVecPion1XiHyp);
+              float p_T_Xic = std::hypot(pVecXi[0] + pVecPion1XiHyp[0], pVecXi[1] + pVecPion1XiHyp[1]);
```
Comment (fgrosa) [inline PWGHF/TableProducer/trackIndexSkimCreator.cxx:3573]:
```suggestion
              float ptXic = RecoDecay::pt(RecoDecay::pVec(pVecXi, pVecPion1XiHyp));
```

File: PWGHF/TableProducer/trackIndexSkimCreator.cxx [hunk 3]
```diff
@@ -3555,17 +3570,23 @@ struct HfTrackIndexSkimCreatorLfCascades {
...
+                if (p_T_Xic >= config.ptMinXicZeroOmegacZeroToXiPiLfCasc) {
+                  SETBIT(hfFlag, aod::hf_cand_casc_lf::DecayType2Prong::XiczeroOmegaczeroToXiPi);
+                  registry.fill(HIST("hRejpTStatusXicZeroOmegacZeroToXiPi"), 1);
+                }
               }
 
               // fill histograms
               if (config.fillHistograms && (TESTBIT(hfFlag, aod::hf_cand_casc_lf::DecayType2Prong::XiczeroOmegaczeroToXiPi))) {
                 registry.fill(HIST("hMassXicZeroOmegacZeroToXiPi"), mass2ProngXiHyp);
+                registry.fill(HIST("hPtXicZeroOmegacZeroToXiPi"), p_T_Xic);
```
Comment (fgrosa) [inline PWGHF/TableProducer/trackIndexSkimCreator.cxx:3589]:
```suggestion
                registry.fill(HIST("hPtXicZeroOmegacZeroToXiPi"), ptXic);
```

File: PWGHF/TableProducer/trackIndexSkimCreator.cxx [hunk 4]
```diff
@@ -3596,27 +3617,38 @@ struct HfTrackIndexSkimCreatorLfCascades {
               std::array<float, 3> pVecCharmBachelor1OmegaHyp = {0.};
               df2.getTrack(0).getPxPyPzGlo(pVecOmega);
               df2.getTrack(1).getPxPyPzGlo(pVecCharmBachelor1OmegaHyp);
+              float p_T_Omegac = std::hypot(pVecOmega[0] + pVecCharmBachelor1OmegaHyp[0], pVecOmega[1] + pVecCharmBachelor1OmegaHyp[1]);
```
Comment (fgrosa) [inline PWGHF/TableProducer/trackIndexSkimCreator.cxx:3620]:
```suggestion
              float ptOmegac = RecoDecay::pt(RecoDecay::pVec(pVecOmega, pVecCharmBachelor1OmegaHyp));
```

File: PWGHF/TableProducer/trackIndexSkimCreator.cxx [hunk 5]
```diff
@@ -3596,27 +3617,38 @@ struct HfTrackIndexSkimCreatorLfCascades {
...
 
               std::array<std::array<float, 3>, 2> arrMomToOmega = {pVecOmega, pVecCharmBachelor1OmegaHyp};
               auto mass2ProngOmegaPiHyp = RecoDecay::m(arrMomToOmega, arrMass2Prong[hf_cand_casc_lf::DecayType2Prong::OmegaczeroToOmegaPi]);
               auto mass2ProngOmegaKHyp = RecoDecay::m(arrMomToOmega, arrMass2Prong[hf_cand_casc_lf::DecayType2Prong::OmegaczeroToOmegaK]);
 
               if (std::abs(casc.mOmega() - massOmega) < config.cascadeMassWindow) {
                 if ((mass2ProngOmegaPiHyp >= config.massOmegaCharmBachelorMin) && (mass2ProngOmegaPiHyp <= config.massOmegaCharmBachelorMax)) {
-                  SETBIT(hfFlag, aod::hf_cand_casc_lf::DecayType2Prong::OmegaczeroToOmegaPi);
+                  registry.fill(HIST("hRejpTStatusOmegacZeroToOmegaPi"), 0);
+                  if (p_T_Omegac >= config.ptMinOmegacZeroToOmegaPiLfCasc) {
```
Comment (fgrosa) [inline PWGHF/TableProducer/trackIndexSkimCreator.cxx:3629]:
```suggestion
                  if (ptOmegac >= config.ptMinOmegacZeroToOmegaPiLfCasc) {
```

File: PWGHF/TableProducer/trackIndexSkimCreator.cxx [hunk 6]
```diff
@@ -3596,27 +3617,38 @@ struct HfTrackIndexSkimCreatorLfCascades {
...
+                    SETBIT(hfFlag, aod::hf_cand_casc_lf::DecayType2Prong::OmegaczeroToOmegaPi);
+                    registry.fill(HIST("hRejpTStatusOmegacZeroToOmegaPi"), 1);
+                  }
                 }
                 if ((mass2ProngOmegaKHyp >= config.massOmegaCharmBachelorMin) && (mass2ProngOmegaKHyp <= config.massOmegaCharmBachelorMax)) {
-                  SETBIT(hfFlag, aod::hf_cand_casc_lf::DecayType2Prong::OmegaczeroToOmegaK);
+                  registry.fill(HIST("hRejpTStatusOmegacZeroToOmegaKa"), 0);
+                  if (p_T_Omegac >= config.ptMinOmegaczeroToOmegaKaLfCasc) {
```
Comment (fgrosa) [inline PWGHF/TableProducer/trackIndexSkimCreator.cxx:3636]:
```suggestion
                  if (ptOmegac >= config.ptMinOmegaczeroToOmegaKaLfCasc) {
```

File: PWGHF/TableProducer/trackIndexSkimCreator.cxx [hunk 7]
```diff
@@ -3596,27 +3617,38 @@ struct HfTrackIndexSkimCreatorLfCascades {
...
+                    registry.fill(HIST("hRejpTStatusOmegacZeroToOmegaKa"), 1);
+                  }
                 }
               }
 
               // fill histograms
               if (config.fillHistograms) {
                 if (TESTBIT(hfFlag, aod::hf_cand_casc_lf::DecayType2Prong::OmegaczeroToOmegaPi)) {
                   registry.fill(HIST("hMassOmegacZeroToOmegaPi"), mass2ProngOmegaPiHyp);
+                  registry.fill(HIST("hPtCutsOmegacZeroToOmegaPi"), p_T_Omegac);
```
Comment (fgrosa) [inline PWGHF/TableProducer/trackIndexSkimCreator.cxx:3647]:
```suggestion
                  registry.fill(HIST("hPtCutsOmegacZeroToOmegaPi"), ptOmegac);
```

File: PWGHF/TableProducer/trackIndexSkimCreator.cxx [hunk 8]
```diff
@@ -3596,27 +3617,38 @@ struct HfTrackIndexSkimCreatorLfCascades {
...
                 }
                 if (TESTBIT(hfFlag, aod::hf_cand_casc_lf::DecayType2Prong::OmegaczeroToOmegaK)) {
                   registry.fill(HIST("hMassOmegacZeroToOmegaK"), mass2ProngOmegaKHyp);
+                  registry.fill(HIST("hPtCutsOmegacZeroToOmegaKa"), p_T_Omegac);
```
Comment (fgrosa) [inline PWGHF/TableProducer/trackIndexSkimCreator.cxx:3651]:
```suggestion
                  registry.fill(HIST("hPtCutsOmegacZeroToOmegaKa"), ptOmegac);
```

File: PWGHF/TableProducer/trackIndexSkimCreator.cxx [hunk 9]
```diff
@@ -3689,17 +3721,23 @@ struct HfTrackIndexSkimCreatorLfCascades {
                   df3.getTrack(0).getPxPyPzGlo(pVec1); // take the momentum at the Xic vertex
                   df3.getTrack(1).getPxPyPzGlo(pVec2);
                   df3.getTrack(2).getPxPyPzGlo(pVec3);
+                  float p_T_Xic3Prong = std::hypot(pVec1[0] + pVec2[0] + pVec3[0], pVec1[1] + pVec2[1] + pVec3[1]);
```
Comment (fgrosa) [inline PWGHF/TableProducer/trackIndexSkimCreator.cxx:3724]:
```suggestion
                  float ptXic3Prong = RecoDecay::pt(RecoDecay::pVec(pVec1, pVec2, pVec3));
```

File: PWGHF/TableProducer/trackIndexSkimCreator.cxx [hunk 10]
```diff
@@ -3689,17 +3721,23 @@ struct HfTrackIndexSkimCreatorLfCascades {
...
 
                   std::array<std::array<float, 3>, 3> arr3Mom = {pVec1, pVec2, pVec3};
                   auto mass3Prong = RecoDecay::m(arr3Mom, arrMass3Prong[hf_cand_casc_lf::DecayType3Prong::XicplusToXiPiPi]);
 
                   if ((std::abs(casc.mXi() - massXi) < config.cascadeMassWindow) && (mass3Prong >= config.massXiPiPiMin) && (mass3Prong <= config.massXiPiPiMax)) {
-                    SETBIT(hfFlag, aod::hf_cand_casc_lf::DecayType3Prong::XicplusToXiPiPi);
+                    registry.fill(HIST("hRejpTStatusXicPlusToXiPiPi"), 0);
+                    if (p_T_Xic3Prong >= config.ptMinXicplusLfCasc) {
```
Comment (fgrosa) [inline PWGHF/TableProducer/trackIndexSkimCreator.cxx:3731]:
```suggestion
                    if (ptXic3Prong >= config.ptMinXicplusLfCasc) {
```

File: PWGHF/TableProducer/trackIndexSkimCreator.cxx [hunk 11]
```diff
@@ -3689,17 +3721,23 @@ struct HfTrackIndexSkimCreatorLfCascades {
...
+                      SETBIT(hfFlag, aod::hf_cand_casc_lf::DecayType3Prong::XicplusToXiPiPi);
+                      registry.fill(HIST("hRejpTStatusXicPlusToXiPiPi"), 1);
+                    }
                   }
 
                   // fill histograms
                   if (config.fillHistograms && (TESTBIT(hfFlag, aod::hf_cand_casc_lf::DecayType3Prong::XicplusToXiPiPi))) {
                     registry.fill(HIST("hMassXicPlusToXiPiPi"), mass3Prong);
+                    registry.fill(HIST("hPtCutsXicPlusToXiPiPi"), p_T_Xic3Prong);
```
Comment (fgrosa) [inline PWGHF/TableProducer/trackIndexSkimCreator.cxx:3740]:
```suggestion
                    registry.fill(HIST("hPtCutsXicPlusToXiPiPi"), ptXic3Prong);
```

**Review by fgrosa — CHANGES_REQUESTED** [review]
Hi @FDUEnrich thanks! The PR looks ok to me in general, I just have a few minor comments (see below)

File: PWGHF/TableProducer/trackIndexSkimCreator.cxx (hunk 2 above)
Comment (vkucera) [reply PWGHF/TableProducer/trackIndexSkimCreator.cxx:3573]: `float ptXic = RecoDecay::pt(pVecXi, pVecPion1XiHyp);`

File: PWGHF/TableProducer/trackIndexSkimCreator.cxx (hunk 4 above)
Comment (vkucera) [reply PWGHF/TableProducer/trackIndexSkimCreator.cxx:3620]: Same here

File: PWGHF/TableProducer/trackIndexSkimCreator.cxx (hunk 9 above)
Comment (vkucera) [reply PWGHF/TableProducer/trackIndexSkimCreator.cxx:3724]: Same here

File: PWGHF/TableProducer/trackIndexSkimCreator.cxx (hunk 2 above)
Comment (FDUEnrich) [reply PWGHF/TableProducer/trackIndexSkimCreator.cxx:3573]: Dear experts, which one is better?

File: PWGHF/TableProducer/trackIndexSkimCreator.cxx [hunk 12]
```diff
@@ -3234,7 +3235,11 @@ struct HfTrackIndexSkimCreatorLfCascades {
     // Selection criteria
     // selections have been set to run2 lambda dedicated cuts
     // selections for cascade have been set to the loosest value between xi and omega
-    // a tolerance has been added to be more conservative
+    // a tolerance has been added to be more conservative  ptMinOmegaczeroToOmegaKaLfCasc ptMinXicZeroOmegacZeroToXiPiLfCasc
```
Comment (vkucera) [inline PWGHF/TableProducer/trackIndexSkimCreator.cxx:3238]: Why did you add `ptMinOmegaczeroToOmegaKaLfCasc ptMinXicZeroOmegacZeroToXiPiLfCasc` here?

Comment (FDUEnrich) [reply PWGHF/TableProducer/trackIndexSkimCreator.cxx:3238]: Dear expert, thank you for comment! I added these four configurable because Fabrizio suggested doing this. We think it's better to cut pT for OmegaczeroToOmegaKa, XicZeroOmegacZeroToXiPi, OmegacZeroToOmegaPi and XicplusToXiPiPi respectively.

Comment (fgrosa) [reply PWGHF/TableProducer/trackIndexSkimCreator.cxx:3238]: @vkucera this is for the production of derived data in Pb-Pb, for which it is unlikely that we can perform measurements to pT=0 and the size is drastically reduced if we skip the low pT

**Review by fgrosa — APPROVED** [review]

File: PWGHF/TableProducer/trackIndexSkimCreator.cxx (hunk 12 above)
Comment (vkucera) [reply PWGHF/TableProducer/trackIndexSkimCreator.cxx:3238]: @FDUEnrich @fgrosa I was referring to that very line with the comment.
